codemod/base.py

```diff
@@ -685,7 +686,18 @@ def _ask_about_patch(patch, editor, default_no):
         _save(patch.path, lines)
     if p in 'eE':
         run_editor(patch.start_position, editor)
+    if p in 'q':
+        sys.exit(0)
 
+def _getch():
+    fd = sys.stdin.fileno()
+    old_settings = termios.tcgetattr(fd)
+    try:
+        tty.setraw(sys.stdin.fileno())
+        ch = sys.stdin.read(1)
+    finally:
+        termios.tcsetattr(fd, termios.TCSADRAIN, old_settings)
```

Comment:
Oh, my bad -- is this `try`/`finally` meant to gate this to just the platforms that support this? If so, then that works for me.